vms/platformvm/api/static_service.go

```diff
@@ -350,59 +304,50 @@ func (*StaticService) BuildGenesis(_ *http.Request, args *BuildGenesisArgs, repl
 		}
 
 		if err := tx.Initialize(txs.GenesisCodec); err != nil {
-			return err
+			return nil, err
 		}
 
 		vdrs.Add(tx)
 	}
 
-	// Specify the chains that exist at genesis.
-	chains := []*txs.Tx{}
-	for _, chain := range args.Chains {
-		genesisBytes, err := formatting.Decode(args.Encoding, chain.GenesisData)
-		if err != nil {
-			return fmt.Errorf("problem decoding chain genesis data: %w", err)
-		}
+	// Specify the chains that exist at genesis
+	chainsTxs := []*txs.Tx{}
+	for _, chain := range chains {
 		tx := &txs.Tx{Unsigned: &txs.CreateChainTx{
 			BaseTx: txs.BaseTx{BaseTx: avax.BaseTx{
-				NetworkID:    uint32(args.NetworkID),
+				NetworkID:    networkID,
 				BlockchainID: ids.Empty,
 			}},
 			SubnetID:    chain.SubnetID,
 			ChainName:   chain.Name,
 			VMID:        chain.VMID,
 			FxIDs:       chain.FxIDs,
-			GenesisData: genesisBytes,
+			GenesisData: chain.GenesisData,
 			SubnetAuth:  &secp256k1fx.Input{},
 		}}
 		if err := tx.Initialize(txs.GenesisCodec); err != nil {
-			return err
+			return nil, err
 		}
 
-		chains = append(chains, tx)
+		chainsTxs = append(chainsTxs, tx)
 	}
 
 	validatorTxs := vdrs.List()
 
 	// genesis holds the genesis state
-	g := genesis.Genesis{
+	g := Genesis{
 		UTXOs:         utxos,
 		Validators:    validatorTxs,
-		Chains:        chains,
-		Timestamp:     uint64(args.Time),
-		InitialSupply: uint64(args.InitialSupply),
-		Message:       args.Message,
+		Chains:        chainsTxs,
+		Timestamp:     time,
+		InitialSupply: initialSupply,
+		Message:       message,
 	}
 
 	// Marshal genesis to bytes
-	bytes, err := genesis.Codec.Marshal(genesis.CodecVersion, g)
-	if err != nil {
-		return fmt.Errorf("couldn't marshal genesis: %w", err)
-	}
-	reply.Bytes, err = formatting.Encode(args.Encoding, bytes)
+	genesisBytes, err := genesis.Codec.Marshal(genesis.CodecVersion, g)
 	if err != nil {
-		return fmt.Errorf("couldn't encode genesis as string: %w", err)
+		return nil, fmt.Errorf("couldn't marshal genesis: %w", err)
 	}
-	reply.Encoding = args.Encoding
-	return nil
+	return genesisBytes, nil
```

Comment:
Similar comment to the AVM - I feel like this function would best return the actual genesis rather than the bytes